ProcessMaker/processmaker
# PM Block: Import/Export [FOUR-7372]

This PR introduces crucial import/export functionality to PM Blocks, enhancing their versatility and usability. It includes the following key changes:

- Updated the incrementStringSeparator variable to handle duplicate items as an array. This enhancement allows for specifying specific string separators for each duplicated attribute. For instance, the 'name' attribute can be separated by a space (" "), while the process_package_key attribute can be separated by a dash ("-").

 ```
 public $handleDuplicatesByIncrementing = ['name', 'process_package_key'];
 public $incrementStringSeparator = [' ', '-'];
 ```

- Implemented the hiding of assets from the UI based on the asset_type column.
- Applied PHP CS FIXER formatting for improved code consistency and readability.


##To Test

1. Create a PM block.
2. Export the PM block from the PM Block listing page.
3. Import the PM block.
4. Verify that the PM block is listed under the PM Block pages.
5. Confirm that any assets attached to the PM block are not displayed within the UI.
6. Ensure the ability to edit the PM block.
7. Run the PM block in a process to validate its functionality.


## Related Tickets & Packages
- PM blocks Epic FOUR-7365


## Code Review Checklist
